spec/text-buffer-spec.coffee

```diff
@@ -2232,12 +2232,31 @@ describe "TextBuffer", ->
             /.{5}/g
           ][random(4)]
 
-          forwardRanges = []
-          backwardRanges = []
-          buffer.scanInRange regex, range, ({range, matchText}) -> forwardRanges.push({range, matchText})
-          buffer.backwardsScanInRange regex, range, ({range, matchText}) -> backwardRanges.unshift({range, matchText})
+          if random(2) > 0
+            forwardRanges = []
+            backwardRanges = []
+            forwardMatches = []
+            backwardMatches = []
 
-          expect(backwardRanges).toEqual(forwardRanges, "Seed: #{seed}")
+            referenceBuffer.scanInRange regex, range, ({range, matchText}) ->
+              forwardMatches.push(matchText)
+              forwardRanges.push(range)
+
+            buffer.backwardsScanInRange regex, range, ({range, matchText}) ->
+              backwardMatches.unshift(matchText)
+              backwardRanges.unshift(range)
+
+            expect(backwardRanges).toEqual(forwardRanges, "Seed: #{seed}")
+            expect(backwardMatches).toEqual(forwardMatches, "Seed: #{seed}")
+          else
+            referenceBuffer = new TextBuffer(text: buffer.getText())
+            referenceBuffer.scanInRange regex, range, ({matchText, replace}) ->
+              replace(matchText + '.')
+
+            buffer.backwardsScanInRange regex, range, ({matchText, replace}) ->
+              replace(matchText + '.')
+
+            expect(buffer.getText()).toBe(referenceBuffer.getText(), "Seed: #{seed}")
```

Comment:
The issue wasn't actually specific to `replace`; The existing tests would occasionally catch it. But I thought I'd add randomized tests for `replace` while I was here anyway.